batch.go

```diff
@@ -160,25 +165,42 @@ func (batch *Batch) ReadMessage() (Message, error) {
 			return
 		},
 	)
+	for batch.conn != nil && offset < batch.conn.offset {
+		if err != nil {
+			goto end
```

Comment:
```suggestion
			break
```
It seems like we're not skipping any code, we just want to break out?

Another option would be to check `for err == nil && ... {` directly in the loop.